tests/system/src/test/java/test/javafx/scene/lighting3D/PointLightAttenuationTest.java

```diff
@@ -0,0 +1,140 @@
+        }
+    }
+
+    @Test
+    public void testAttenuation() {
+        Util.runAndWait(() -> {
+            // Since there appears to be a bug in snapshot with subscene, we are taking a snapshot of the scene and not
+            // the box, so the center of the box will be at the top left, (0, 0), of the image, and the light is
+            // straight in front. Without attenuation, at (0, 0) it will give its full color. At (SAMPLE_DIST, 0) and
+            // LIGTH_DIST == SAMPLE_DIST, it will give cos(45) = 1/sqrt(2) of its color.
+            var snapshot = box.getScene().snapshot(null);
+            double nonAttenBlueCenter = snapshot.getPixelReader().getColor(0, 0).getBlue();
+            double nonAttenBlueDiag = snapshot.getPixelReader().getColor(SAMPLE_DIST, 0).getBlue();
+            assertEquals("Wrong color value", 1, nonAttenBlueCenter, DELTA);
+            assertEquals("Wrong color value", 1/Math.sqrt(2), nonAttenBlueDiag, DELTA);
+
+            double diagDist = Math.sqrt(LIGTH_DIST * LIGTH_DIST + SAMPLE_DIST * SAMPLE_DIST);
+
+            light.setLinearAttenuation(0.01);
+            double attnCenter = 1 / (1 + 0.01 * LIGTH_DIST);
+            double attnDiag = 1 / (1 + 0.01 * diagDist);
+            snapshot = box.getScene().snapshot(null);
+            double attenBlueCenter = snapshot.getPixelReader().getColor(0, 0).getBlue();
+            double attenBlueDiag = snapshot.getPixelReader().getColor(SAMPLE_DIST, 0).getBlue();
+            assertEquals("Wrong color value", nonAttenBlueCenter * attnCenter, attenBlueCenter, DELTA);
+            assertEquals("Wrong color value", nonAttenBlueDiag * attnDiag, attenBlueDiag, DELTA);
+
+            light.setLinearAttenuation(0);
+            light.setQuadraticAttenuation(0.01);
+            attnCenter = 1 / (1 + 0.01 * LIGTH_DIST * LIGTH_DIST);
+            attnDiag = 1 / (1 + 0.01 * diagDist * diagDist);
+            snapshot = box.getScene().snapshot(null);
+            attenBlueCenter = snapshot.getPixelReader().getColor(0, 0).getBlue();
+            attenBlueDiag = snapshot.getPixelReader().getColor(SAMPLE_DIST, 0).getBlue();
+            assertEquals("Wrong color value", nonAttenBlueCenter * attnCenter, attenBlueCenter, DELTA);
+            assertEquals("Wrong color value", nonAttenBlueDiag * attnDiag, attenBlueDiag, DELTA);
+
+            light.setQuadraticAttenuation(0);
+            light.setMaxRange((LIGTH_DIST + diagDist) / 2);
+            snapshot = box.getScene().snapshot(null);
+            nonAttenBlueCenter = snapshot.getPixelReader().getColor(0, 0).getBlue();
+            nonAttenBlueDiag = snapshot.getPixelReader().getColor(SAMPLE_DIST, 0).getBlue();
+            assertEquals("Wrong color value, should be in range", 1, nonAttenBlueCenter, DELTA);
+            assertEquals("Wrong color value, should be out of range", 0, nonAttenBlueDiag, DELTA);
+        });
+    }
+
+    @AfterClass
+    public static void teardown() {
+        Platform.runLater(() -> {
```

Comment:
I forgot to mention that the `@AfterClass` method is still run even if the call to `assumeTrue` in the `@BeforeClass` method causes the test to be skipped. You will need to repeat the check (or set a flag in `initFX` and check that flag) here and skip the cleanup if the initialization was skipped (using an `if` check, not repeating the `assumeTrue`). Otherwise the exception that is thrown here will cause the test to fail.

You can test this by forcing the SW pipeline. I did it like this:

```
_JAVA_OPTIONS="-Dprism.order=sw" gradle --info -PFULL_TEST=true :systemTests:test --tests PointLightAttenuationTest
```

Reply:
The test fails even with this addition. I found that I need to use `@Before` to run the check.